static/bidder-params/appnexus.json

```diff
@@ -0,0 +1,65 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Appnexus Adapter Params",
+  "description": "A schema which validates params accepted by the AppNexus adapter",
+
+  "type": "object",
+  "properties": {
+    "placementId": {
+      "type": "integer",
+      "description": "An ID which identifies this placement"
+    },
+    "invCode": {
+      "type": "string",
+      "description": "The placement code, aka inventory code, for this placement."
+    },
+    "member": {
+      "type": "string",
+      "description": "The member ID from AppNexus."
+    },
+    "keywords": {
+      "type": "array",
+      "minItems": 1,
+      "items": {
+        "type": "object",
+        "description": "A key with one or more values associated with it. These are used in buy-side segment targeting.",
+        "properties": {
+          "key": {
+            "type": "string"
+          },
+          "value": {
+            "type": "array",
+            "minItems": 1,
+            "items": {
+              "type": "string"
+            }
+          }
+        },
+        "required": ["key", "value"]
+      }
+    },
+    "trafficSourceCode": {
+      "type": "string",
+      "description": "Specifies the third-party source of this impression."
+    },
+    "reserve": {
+      "type": "number",
+      "description": "The minimium acceptable bid, in CPM, using US Dollars"
+    },
+    "position": {
+      "type": "string",
+      "enum": ["above", "below"],
+      "description": "Specifies the ad unit as above or below the fold"
+    }
+  },
+
+  "oneOf": [{
+    "required": ["placementId"]
+  }, {
+    "required": ["invCode", "member"]
+  }],
+
+  "not": {
+    "required": ["placementId", "invCode", "member"]
```

Comment:
What does this mean exactly when one of these three _is_ required?

Reply:
"not" just means "the JSON blob must _not_ match the following schema."

This block rejects any JSON payloads which _do_ define all three. For more info, see https://spacetelescope.github.io/understanding-json-schema/reference/combining.html#not

Reply:
Oh interesting. Just curious.. why do we not want all three to be defined?

Reply:
you know.... I don't actually know. I wrote these to describe the current API, not to define a new one.

PBS doesn't have docs... but [the Prebid.JS ones](http://prebid.org/dev-docs/bidders.html#appnexus) sounded like `invCode/member` was an _alternative_ for `placementId`.

I asked the team to clarify whether a request with all 3 was valid, and @mkendall07 said "no"... so that's what I wrote into the schema.